homeassistant/components/mikrotik/hub.py

```diff
@@ -354,21 +346,9 @@ async def async_add_options(self):
                 self.config_entry, data=data, options=options
             )
 
-    async def request_update(self):
-        """Request an update."""
-        if self.progress is not None:
-            await self.progress
-            return
-
-        self.progress = self.hass.async_create_task(self.async_update())
-        await self.progress
-
-        self.progress = None
-
     async def async_update(self):
         """Update Mikrotik devices information."""
-        await self.hass.async_add_executor_job(self._mk_data.update)
-        async_dispatcher_send(self.hass, self.signal_update)
+        await self.hass.async_add_executor_job(self._mk_data.update_devices)
 
     async def async_setup(self):
```

Comment:
In a future PR it would be good to move to into `__init__.py` since it makes it harder to see what is going on at startup with having it on the coordinator object